gammapy/datasets/tests/test_utils.py

```diff
@@ -35,3 +44,33 @@ def test_apply_edisp(region_map_true):
     assert e_reco.unit == "TeV"
     assert m.geom.axes[0].name == "energy"
     assert_allclose(e_reco[[0, -1]].value, [1, 10])
+
+
+@requires_data()
+def test_dataset_split():
+
+    template_diffuse = TemplateSpatialModel.read(
+        filename="$GAMMAPY_DATA/fermi-3fhl-gc/gll_iem_v06_gc.fits.gz",
+        normalize=False,
+    )
+
+    diffuse_iem = SkyModel(
+        spectral_model=PowerLawNormSpectralModel(),
+        spatial_model=template_diffuse,
+        name="diffuse-iem",
+    )
+
+    dataset = MapDataset.read("$GAMMAPY_DATA/fermi-3fhl-gc/fermi-3fhl-gc.fits.gz")
+    dataset.models = Models([diffuse_iem])
+
+    width = 4 * u.deg
+    margin = 2 * u.deg
+    datasets = split_dataset(dataset, width, margin)
+    assert len(datasets) == 15
+    assert len(datasets.models) == 1
+
+    datasets = split_dataset(
+        dataset, width=4 * u.deg, margin=2 * u.deg, split_templates=True
+    )
+    assert len(datasets.models) == len(datasets)
+    assert len(datasets.parameters.free_parameters) == 1
```

Comment:
Some tests might be needed for the 2 masks

Reply:
Can you test also that the models are really cutout?

Reply:
done